src/dashboard/tests/test_transfer.py

```diff
@@ -0,0 +1,52 @@
+import os
+
+from django.core.urlresolvers import reverse
+from django.test import TestCase
+from django.test.client import Client
+
+from components import helpers
+from main.models import DublinCore
+
+
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+
+
+class TestTransferViews(TestCase):
+    fixture_files = ['test_user.json', 'transfer.json']
+    fixtures = [os.path.join(THIS_DIR, 'fixtures', p) for p in fixture_files]
+
+    def setUp(self):
+        self.client = Client()
+        self.client.login(username='test', password='test')
+        helpers.set_setting('dashboard_uuid', 'test-uuid')
+
+    def test_metadata_edit(self):
+        """Test the metadata form of a transfer"""
+        transfer_uuid = '3e1e56ed-923b-4b53-84fe-c5c1c0b0cf8e'
+        url = reverse(
+            'components.transfer.views.transfer_metadata_edit',
+            args=[transfer_uuid],
+        )
+        response = self.client.post(url, {
+            'title': 'My First Transfer',
+            'is_part_of': '1234',
+            'creator': 'The Creator',
+            'subject': 'A Subject',
+            'description': 'Transfer description',
+            'publisher': 'The Publisher',
+            'contributor': 'A contributor',
+            'date': '2019-01-01',
+        })
+        # Verify changes
+        transfer_metadata = DublinCore.objects.get(
+            metadataappliestoidentifier=transfer_uuid)
+        assert transfer_metadata.title == 'My First Transfer'
+        assert transfer_metadata.is_part_of == 'AIC#1234'
+        assert transfer_metadata.creator == 'The Creator'
+        assert transfer_metadata.subject == 'A Subject'
+        assert transfer_metadata.description == 'Transfer description'
+        assert transfer_metadata.publisher == 'The Publisher'
+        assert transfer_metadata.contributor == 'A contributor'
+        assert transfer_metadata.date == '2019-01-01'
+        # Verify form redirects to the metadata list after saving
+        assert response.url == 'http://testserver/transfer/{}/metadata/'.format(transfer_uuid)
```

Comment:
Should that value use `reverse` to?

Reply:
`reverse` doesn't ouput the `http://testserver/` part iirc?

Reply:
Maybe:

```
        redirect_url = 'http://testserver{}'.format(reverse(
            'components.transfer.views.transfer_metadata_list',
            args=[transfer_uuid],
        ))
```